File: CodeGeneratorMVC/oldClasses/ProjectVariable.vb
Option Strict On
<Serializable()> Public Class ProjectVariable
    Private _ID As Integer
    Private _Name As String
    Public Property ID() As Integer
        Get
            Return _ID
        End Get
        Set(ByVal value As Integer)
            _ID = value
        End Set
    End Property
    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property
    Public ReadOnly Property NameBasedOnID() As String
        Get
            If Name = StaticVariables.Instance.NoAssociatedNameSpaceString Then
                Return ""
            Else
                Return _Name
            End If

        End Get
    End Property
    Public Sub New()

    End Sub
    Public Sub New(ByVal ProjectVariableID As Integer, ByVal ProjectVariableName As String)
        _ID = ProjectVariableID
        _Name = ProjectVariableName
    End Sub
    Public Overrides Function ToString() As String
        Return _Name
    End Function
    Public ReadOnly Property Self() As ProjectVariable
        Get
            Return Me
        End Get
    End Property

    Public Overrides Function Equals(ByVal obj As Object) As Boolean
        If TypeOf obj Is ProjectVariable Then
            Return Me = CType(obj, ProjectVariable)
        End If
        Return False
    End Function
    Public Overloads Shared Operator =(ByVal pv1 As ProjectVariable, ByVal pv2 As ProjectVariable) As Boolean
        If pv1 Is Nothing And pv2 Is Nothing Then
            Return True
        End If
        If pv1 Is Nothing OrElse pv2 Is Nothing Then
            Return False
        End If
        Return pv1.ID = pv2.ID
    End Operator
    Public Overloads Shared Operator <>(ByVal pv1 As ProjectVariable, ByVal pv2 As ProjectVariable) As Boolean
        Return Not pv1 = pv2
    End Operator



End Class

File: CodeGeneratorMVC/Models/ProjectVariable.cs
using System;

[Serializable()]
public class ProjectVariable {
    private int _ID;
    private string _Name;
    public int ID {
        get {
            return _ID;
        }
        set {
            _ID = value;
        }
    }
    public string Name {
        get {
            return _Name;
        }
        set {
            _Name = value;
        }
    }
    public string NameBasedOnID {
        get {
            if (Name == StaticVariables.Instance.NoAssociatedNameSpaceString)
                return "";
            else
                return _Name;
        }
    }
    public ProjectVariable() {
    }
    public ProjectVariable(int ProjectVariableID, string ProjectVariableName) {
        _ID = ProjectVariableID;
        _Name = ProjectVariableName;
    }
    public override string ToString() {
        return _Name;
    }
    public ProjectVariable Self {
        get {
            return this;
        }
    }

    public override bool Equals(object obj) {
        if (obj is ProjectVariable)
            return ReferenceEquals(this,(ProjectVariable)obj);
        return false;
    }
    public static bool operator ==(ProjectVariable pv1, ProjectVariable pv2) {
        if (ReferenceEquals(pv1,null) & ReferenceEquals(pv2,null))
            return true;
        if (ReferenceEquals(pv1, null) || ReferenceEquals(pv2, null))
            return false;
        return pv1.ID == pv2.ID;
    }
    public static bool operator !=(ProjectVariable pv1, ProjectVariable pv2) {
        return !(pv1 == pv2);
    }
}
